misc/python/materialize/scalability/plot/plot.py

```diff
@@ -76,16 +77,41 @@ def boxplot_latency_per_connections(
         df_details_cols.CONCURRENCY
     ].unique()
 
-    subplots = figure.subplots(1, len(concurrencies), sharey=True)
+    endpoint_names = df_details_by_endpoint_name.keys()
+    use_short_names = len(endpoint_names) > 2
+
+    num_rows = math.ceil(len(concurrencies) / 3)
+    num_cols = math.ceil(len(concurrencies) / num_rows)
+    use_no_grid = num_rows == 1 and num_cols == 1
+    use_single_row = num_rows == 1 and num_cols > 1
+
+    subplots = figure.subplots(num_rows, num_cols, sharey=False)
 
     for concurrency_index, concurrency in enumerate(concurrencies):
-        plot: Axes = subplots[concurrency_index]
+        if use_no_grid:
+            plot: Axes = subplots
+            is_in_first_column = True
+        elif use_single_row:
+            plot: Axes = subplots[concurrency_index]
+            is_in_first_column = concurrency_index == 0
+        else:
+            row = math.floor(concurrency_index / num_cols)
+            column = concurrency_index % num_cols
+            plot: Axes = subplots[row][column]
+            is_in_first_column = column == 0
+
+        assert type(plot) == Axes
```

Comment:
The type returned by `figure.subplots(num_rows, num_cols, sharey=False)` depends on the number of rows and columns... 🤦‍♂️